src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -0,0 +1,355 @@
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (6,29): error CS1061: 'C' does not contain a definition for 'field' and no accessible extension method 'field' accepting a first argument of type 'C' could be found (are you missing a using directive or an assembly reference?)
+                //         get { return _other.field; }
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "field").WithArguments("C", "field").WithLocation(6, 29),
+                // (7,19): info CS9258: 'field' is a contextual keyword in property accessors starting in language version preview. Use '@field' instead.
+                //         set { _ = field; }
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "field").WithArguments("field", "preview").WithLocation(7, 19));
+        }
+
+        [Fact]
+        public void FieldReference_02()
+        {
+            string source = """
+                class C
+                {
+                    C P
+                    {
+                        get { return null; }
+                        set { field = value.field; }
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (6,15): info CS9258: 'field' is a contextual keyword in property accessors starting in language version preview. Use '@field' instead.
+                //         set { field = value.field; }
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "field").WithArguments("field", "preview").WithLocation(6, 15),
+                // (6,29): error CS1061: 'C' does not contain a definition for 'field' and no accessible extension method 'field' accepting a first argument of type 'C' could be found (are you missing a using directive or an assembly reference?)
+                //         set { field = value.field; }
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "field").WithArguments("C", "field").WithLocation(6, 29));
+        }
+
+        [Fact]
+        public void FieldReference_03()
+        {
+            string source = """
+                class C
+                {
+                    int P
+                    {
+                        get { return field; }
+                        set { _ = this is { field: 0 }; }
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (5,22): info CS9258: 'field' is a contextual keyword in property accessors starting in language version preview. Use '@field' instead.
+                //         get { return field; }
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "field").WithArguments("field", "preview").WithLocation(5, 22),
```

Comment:
I believe we want to report this info-level diagnostic always, even with language version preview, to allow users to discover where there may be a breaking change. But perhaps we should only report the diagnostic when `field` would otherwise bind to a member other than the synthesized field. cc @jaredpar

Reply:
Added a PROTOTYPE comment to `Binder.ReportFieldContextualKeywordConflict`.